src/nvim/lua/executor.c

```diff
@@ -2313,3 +2313,58 @@ void nlua_init_defaults(void)
     os_errmsg("\n");
   }
 }
+
+bool nlua_func_exists(const char *lua_funcname)
+{
+  lua_State *const L = global_lstate;
+
+  // Copy the Lua function name to avoid modifying the original string
+  char *funcname_copy = xstrdup(lua_funcname);
+
+  // Replace escaped characters in the function name
+  char *p = funcname_copy;
+  while (*p) {
+    if (*p == '\\' && *(p + 1) != '\0') {
+      memmove(p, p + 1, strlen(p));
+    }
+    p++;
+  }
+
+  // Check if the function starts with 'require'
+  if (strncmp(funcname_copy, "require(", 8) == 0) {
+    // Extract the module name from 'require' call
+    char *module_name = strchr(funcname_copy, '(') + 1;
+    char *closing_parenthesis = strchr(module_name, ')');
+
+    if (module_name && closing_parenthesis) {
+      *closing_parenthesis = '\0';  // Null-terminate the module name
+      lua_getglobal(L, module_name);
```

Comment:
The value pushed onto the stack is not used? This will always reach the `return true` statement below.
And `module_name` includes quotes here.